### ISSUE TITLE
Transition problem for RunPeriod - bad End Year

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6933]
Issue overview
--------------
Transition is inserting years when not necessary, and also end year is one too many in this case.

#### Before
```
  RunPeriod,
    RP1,                        !- Name
    1,                       !- Begin Month
    14,                      !- Begin Day of Month
    1,                       !- End Month
    14,                      !- End Day of Month
    Tuesday,                 !- Day of Week for Start Day
    Yes,                     !- Use Weather File Holidays and Special Days
    Yes,                     !- Use Weather File Daylight Saving Period
    No,                      !- Apply Weekend Holiday Rule
    Yes,                     !- Use Weather File Rain Indicators
    Yes,                     !- Use Weather File Snow Indicators
    1;                       !- Number of Times Runperiod to be Repeated
```
#### After
```
  RunPeriod,
    RP1,                     !- Name
    1,                       !- Begin Month
    14,                      !- Begin Day of Month
            2014,            !- Begin Year
    1,                       !- End Month
    14,                      !- End Day of Month
            2015,            !- End Year
    Tuesday,                 !- Day of Week for Start Day
    Yes,                     !- Use Weather File Holidays and Special Days
    Yes,                     !- Use Weather File Daylight Saving Period
    No,                      !- Apply Weekend Holiday Rule
    Yes,                     !- Use Weather File Rain Indicators
    Yes,                     !- Use Weather File Snow Indicators
    ;                        !- Treat Weather as Actual
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus v9.0.0-IDDFreeze d62a9089b1
 - Noted while testing #6757

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file - attached [6750-DOAToUnitarySystem_SPControl-Before_V890.idf.txt](https://github.com/NREL/EnergyPlus/files/2325770/6750-DOAToUnitarySystem_SPControl-Before_V890.idf.txt)

 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by jasondegraw** [comment]
Fixed on bugfix branch

**Comment by mjwitte** [comment]
Closed via #6941
